### ISSUE TITLE
openssl 1.1 "make" failed

### ISSUE THREAD

**Issue by siddharthupmanyu** [skoobe/riofs#147]
Logs : 
gcc  -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/fuse -D_FILE_OFFSET_BITS=64 -I/usr/include/libxml2      -g -O2   -o riofs riofs-log.o riofs-dir_tree.o riofs-rfuse.o riofs-http_connection.o riofs-http_connection_dir_list.o riofs-bucket_client.o riofs-client_pool.o riofs-file_io_ops.o riofs-cache_mng.o riofs-stat_srv.o riofs-utils.o riofs-conf.o riofs-range.o riofs-main.o  -lglib-2.0 -lfuse -pthread -lxml2 -lcrypto  -levent      -lssl 

**riofs-main.o: In function `application_destroy':
riofs/src/main.c:582: undefined reference to `CRYPTO_mem_leaks_fp'
collect2: error: ld returned 1 exit status**

make[1]: *** [Makefile:374: riofs] Error 1
make[1]: Leaving directory 'riofs/src'
make: *** [Makefile:455: all-recursive] Error 1

**Comment by aaronpederson** [comment]
Yeah, exact same log information for myself as well.

**Comment by aaronpederson** [comment]
@siddharthupmanyu 
So I did some digging tonight and I found the cause and solution on my side, hopefully this assists you and this issue can be closed.
In my case the server in question was running Ubuntu 14.04 and had PHP5.6 installed, recently the [php repository](https://launchpad.net/~ondrej/+archive/ubuntu/php) was updated to use the recent OpenSSL update (1.1.*), while my system was utilizing 1.0.* as a higher priority.
So I version pinned libssl-dev in my server and re-ran the riofs commands.... success.
